example/tweet-performance-tracker/main.bal

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+import ballerina/time;
+import ballerinax/twitter;
+
+configurable string accessToken = ?;
+
+public function main() returns error? {
+
+    // Initialize the client
+    twitter:Client twitter = check new ({
+        auth: {
+            token: accessToken
+        }
+    });
+
+    // Define the username to track and the time range
+    string username = "";
+    time:Utc utcTimeNow = time:utcNow();
+    time:Utc utcTimeBeforeMonth = time:utcAddSeconds(utcTimeNow, -2592000);
+
+    // Initialize tweetsPerformingList
+    twitter:Tweet[] tweetsPerformingList = [];
+
+    // Retrieve user by username
+    twitter:Get2UsersByUsernameUsernameResponse responseUser = check twitter->/users/'by/username/[username]();
+    twitter:User? user = responseUser.data;
+
+    if (user is twitter:User) {
+        twitter:UserId userId = user.id;
+
+        // Retrieve tweets posted by the user in the past month
+        twitter:Get2UsersIdTweetsResponse responseUserTweets = check twitter->/users/[userId]/tweets({
+            start_time: time:utcToString(utcTimeBeforeMonth),
+            end_time: time:utcToString(utcTimeNow)
+        });
+        twitter:Tweet[] tweets = responseUserTweets.data ?: [];
+
+        // Define query parameters
+        twitter:FindTweetByIdQueries queries = {
+            tweet\.fields: ["public_metrics"]
+        };
+
+        // Retrieve detailed metrics for each tweet
+        foreach twitter:Tweet tweet in tweets {
+            twitter:TweetId tweetId = tweet.id ?: "";
+            twitter:Get2TweetsIdResponse responseTweet = check twitter->/tweets/[tweetId](queries = queries);
+            twitter:Tweet? tweetData = responseTweet.data;
+
+            if tweetData != null {
+                tweet.public_metrics = tweetData.public_metrics;
+                tweetsPerformingList.push(tweet);
+            }
+        }
+    }
+
+    // Sort tweetsPerformingList by like_count using bubble sort
+    int n = tweetsPerformingList.length();
```

Comment:
Shall we use meaningful variable names ? And before that please check whether this variable is necessary. IMO we could have just written the for-loop like this:
```ballerina
    foreach int i in 0 ... tweetsPerformingList.length() - 2 {
        // logic should come here
    }
```

Reply:
Solved in : [Fix minor code formatting in examples](https://github.com/ballerina-platform/module-ballerinax-twitter/pull/111/commits/c955f46ceefb7d53336e8f627f62a7edc126e76d)